console/v1alpha1/types_console_plugin.go

```diff
@@ -26,42 +26,82 @@ type ConsolePluginSpec struct {
 	// +kubebuilder:validation:Required
 	// +kubebuilder:validation:MinLength=1
 	// +optional
-	DisplayName string `json:"displayName"`
+	DisplayName string `json:"displayName,omitempty"`
 	// service is a Kubernetes Service that exposes the plugin using a
 	// deployment with an HTTP server. The Service must use HTTPS and
 	// Service serving certificate. The console backend will proxy the
 	// plugins assets from the Service using the service CA bundle.
 	// +kubebuilder:validation:Required
 	// +required
 	Service ConsolePluginService `json:"service"`
-	// proxy is a list of Services that the plugin needs to connect to.
+	// proxy is a list of proxies that describe various service type
+	// to which the plugin needs to connect to.
 	// +kubebuilder:validation:Optional
 	// +optional
-	Proxy ConsolePluginProxy `json:"proxy"`
+	Proxy []ConsolePluginProxy `json:"proxy,omitempty"`
 }
 
 // ConsolePluginProxy holds information on various service types
 // to which console's backend will proxy the plugin's requests.
 type ConsolePluginProxy struct {
-	// services is a list of in-cluster Services that the plugin
-	// will connect to. The Service must use HTTPS. Console backend
-	// exposes the following endpoint in order to proxy communication
-	// between the plugin and the Service:
+	// type is the type of the console plugin's proxy. Currently only "Service" is supported.
+	// +kubebuilder:validation:Required
+	// +required
```

Comment:
Add a pattern so only `Service` is accepted.

Reply:
Thats already added on the `ConsolePluginProxyType` type below